src/main/scala/io/applicative/datastore/util/reflection/ReflectionHelper.scala

```diff
@@ -217,16 +225,31 @@ private[datastore] trait ReflectionHelper extends DateTimeHelper {
       case DatastoreLatLongClassName => LatLng.of(0.0, 0.0)
       case DatastoreBlobClassName => Blob.copyFrom(Array[Byte]())
       case OptionClassName => None
-      case fieldName => throw UnsupportedFieldTypeException(fieldName)
+      case _ => null
```

Comment:
Does this cause an NPE when casting later?

Reply:
I can't imagine when it could happen. This method is used just to create an instance of certain class with some default values. For unknown types of data the default value is null. But this null is replaced later with the method `setActualFieldValues`, so it won't appear in the final object.